File: web_shop_backend/src/main/java/com/lap/web_shop_backend/model/Customer.java
package com.lap.web_shop_backend.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Customer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String email;

    private String phone_number;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "billing_address")
    private Address billing_address;

    @OneToMany(mappedBy = "customer")
    @JsonIgnore
    private List<Shipment> shipments;

    @OneToOne(mappedBy = "customer")
    @JsonIgnore
    private Account account;
}

File: web_shop_frontend/src/models/Customer.ts
import Address from './Address'

export default class Customer {
  id: number
  email: String
  phone_number: String
  billing_address: Address

  constructor(data: any) {
    this.id = data.id
    this.email = data.email
    this.phone_number = data.phone_number
    this.billing_address = new Address(data.billing_address)
  }
}
